pkg/blockbuilder/blockbuilder.go

```diff
@@ -492,6 +543,12 @@ func (b *BlockBuilder) consumePartition(
 
 		recIter := fetches.RecordIter()
 		for !recIter.Done() {
+			metricUpdate++
+			if metricUpdate%10000 == 0 {
+				consumerLagMetric.Set(float64(lag))
+				metricUpdate = 0
+			}
```

Comment:
I'm also no sure there is a need to track the lag from inside the cycle. The lag isn't static, i.e. while cycle is going on Kafka producer keep producing messages, thus I think this is misleading.

What we care about from the inside of the block builder's cycle is how fast we consume + process records.

Also note that we can monitor per-consumer group lag from Kafka itself, as part of how we observe Kafka cluster.